src/components/TeamMemberTasks/TeamMemberTasks.jsx

```diff
@@ -15,257 +19,223 @@ import { getUserProfile } from '../../actions/userProfile';
             };
+            allMembers = allMembers.concat(data[i].data);
           }
 
-          // fetch full user profile for each team member
+          console.log('after teams call');
+        
+          // fetch all time entries for current week for all members
+          const uniqueMembers = _.uniqBy(allMembers, '_id');
           uniqueMembers.forEach((member) => {
-            userProfilePromises.push(
-              httpService.get(ENDPOINTS.USER_PROFILE(member._id)).catch((err) => {}),
+            const fromDate = moment()
+              .tz('America/Los_Angeles')
+              .startOf('week')
+              .subtract(0, 'weeks');
+        
+            const toDate = moment()
+              .tz('America/Los_Angeles')
+              .endOf('week')
+              .subtract(0, 'weeks');
+        
+            memberTimeEntriesPromises.push(
+              httpService.get(ENDPOINTS.TIME_ENTRIES_PERIOD(member._id, fromDate, toDate)).catch((err) => { }),
             );
           });
-          Promise.all(userProfilePromises).then(async (data) => {
-            try {
-              for (let i = 0; i < uniqueMembers.length; i++) {
-                const user = uniqueMembers[i];
-                const userLeaderBoardData = data.find((member) => member.data._id === user._id);
-                let userWeeklyCommittedHours = 0;
-                if (userLeaderBoardData) {
-                  userWeeklyCommittedHours = userLeaderBoardData.data.weeklyComittedHours;
-                }
-                uniqueMembers[i] = {
-                  ...uniqueMembers[i],
-                  weeklyCommittedHours: userWeeklyCommittedHours,
-                };
-              }
 
-              // for each task, must fetch the projectId of its wbs in order to generate appropriate link
-              // currently fetches all projects, should consider refactoring if number of projects increases
-              const WBSRes = await httpService.get(ENDPOINTS.WBS_ALL).catch((err) => {
-                if (err.status === 401) {
-                  loggedOut = true;
-                }
-              });
-              const allWBS = WBSRes.data;
+          Promise.all(memberTimeEntriesPromises).then((data) => {
+            console.log('time entries', data);
+            // merge time entries into each user obj
+            for (let i = 0; i < uniqueMembers.length; i++) {
+              uniqueMembers[i] = {
+                ...uniqueMembers[i],
+                timeEntries: data[i].data,
+              };
+            }
 
-              // calculate hours done in current week and add to user obj for ease of access
+            console.log('members: ', uniqueMembers);
+        
+            // fetch all tasks for each member
+            uniqueMembers.forEach((member) => {
+              teamMemberTasksPromises.push(httpService.get(ENDPOINTS.TASKS_BY_USERID(member._id)).catch((err) => { if (err.status !== 401) { console.log(err); } }));
```

Comment:
We should call one endpoint send that endpoint the list of user_id's and that endpoint should return back the tasks for those user ids with one call to mongo, not a call to mongo for each and every member.